### PR TITLE
Add extra tests for the non-det issue with constants.

### PR DESCRIPTION
**Pull request by NlightNFotis**
This is essentially #1678, ported from `goto-analyzer-develop` to `develop`. 

Paging @martin-cs for a review.

### CODE REVIEW

File: regression/goto-instrument/no_nondet_const/test.desc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CORE
+main.c
+--nondet-static a.out
```
Comment (tautschnig) [inline regression/goto-instrument/no_nondet_const/test.desc]: I believe you need to drop "a.out" from the above to make tests work.

**Review by tautschnig — CHANGES_REQUESTED** [review]

**Comment by NlightNFotis** [comment]
@tautschnig I have implemented the fix and it now passes CI. Should be ready to be merged now.

**Review by tautschnig — APPROVED** [review]

**Comment by tautschnig** [comment]
The changes as such are fine, thanks. Yet could I please ask for the commit message to be amended to answer the “Why?” question? I had to click through three pull requests to even understand what this was good for. And then none of that info will be in the git history.

If I have one Christmas wish for this project, it is that everyone reads and adheres to https://chris.beams.io/posts/git-commit/. Happy Holidays!

**Comment by martin-cs** [comment]
@NlightNFotis I'd really like to merge this; please can you sort this out?

**Comment by NlightNFotis** [comment]
@tautschnig @martin-cs  Is the new commit message better?

**Comment by tautschnig** [comment]
Thanks a lot for the update! Except for a typo ("teh") that seems perfect. I'll merge once CI completes.

**Comment by NlightNFotis** [comment]
@tautschnig I also fixed the typo. Now it should be okay.

**Pull request merged**